src/domains/Townhall/TownhallSettings/TownhallSettings.tsx

```diff
@@ -50,6 +50,8 @@ export default function TownhallSettings() {
     const [save, isLoading] = useEndpoint(() => configureTownhall(townhall._id, state), {
         onSuccess: () => {
             snack('Saved!');
+            
+            setTownhall({...townhall, settings:state});
```

Comment:
This should probably use `setTownhall(prev => ({ ...townhall, settings: state}));`. It's a bit hard to explain over text, but I could do a call with you to explain.